README.md

```diff
@@ -1,98 +1,168 @@
+      - [Using the `backuptar` package](#using-the-backuptar-package)
+      - [Using a CLI command](#using-a-cli-command)
+    - [Passing the backup `tar` file](#passing-the-backup-tar-file)
 
 ![diagram](img/snapshotter-diagram.png)
 
 ## Build snapshotter image
 
-To build the image using this public repository as context you can use the
-following command:
+```yaml
+docker build -t snapshotter:v0.2.0 github.com/NethermindEth/docker-volumes-snapshotter.git#v0.2.0
+```
+
+## Backup
+
+To backup volumes from a Docker container use the `backup` command, bind the volumes, configuration file and the `backup.tar` file.
 
 ```bash
-docker build -t docker-volumes-snapshotter github.com/NethermindEth/docker-volumes-snapshotter.git
+docker run \
+  --rm \
+  --volumes-from <container> \
+  -v $(pwd)/backup.tar:/backup.tar \
+  -v $(pwd)/config.yml:/config.yml \
+  eigenlayer-snapshotter:v0.2.0 backup
 ```
 
-If you prefer, you can use the Dockerfile from this repository directly cloning
-the repository first:
+## Restore
+
+To restore volumes of a Docker container use the `restore` command, bind volumes, configuration file and the `backup.tar` file.
 
 ```bash
-git clone github.com/NethermindEth/docker-volumes-snapshotter.git
-cd docker-volumes-snapshotter
-docker build -t docker-volumes-snapshotter .
+docker run \
+  --rm \
+  --volumes-from <container> \
+  -v $(pwd)/backup.tar:/backup.tar \
+  -v $(pwd)/config.yml:/config.yml \
+  eigenlayer-snapshotter:v0.2.0 restore
 ```
 
-## Running the snapshotter
+## Configuration file
 
-To run the snapshotter a configuration file is required. This file is a YAML file with the following structure:
+### Passing the configuration file
 
-```yaml
-prefix: "prefix/path/to/store/the/backup"
-out: "/path/to/backup.tar"
-volumes:
-  - "/path/to/volume1"
-  - "/path/to/volume2"
+The snapshotter process requires the configuration file to be located at the `/config.yml` path inside the container. Therefore, the recommended way to pass the configuration is by mounting the following volume:
+
+```text
+--volume <path-to-config>:/config.yml
 ```
 
-The snapshotter tries to append the volumes to the backup tar file. If the tar file does not exists, or is empty, an error is raised. To initialize a proper empty tar file run the following command:
+Replace `<path-to-config>` with the absolute path to the configuration file on the host machine.
 
-```bash
-tar -cvf backup.tar --files-from /dev/null
-```
+### Configuration format
 
-You can run the snapshotter different times targeting the same output tar file using different prefix paths. This is useful to create backups of different containers in the same tar file.
+The snapshotter does not need too many configurations, only two options are necessary:
+
+1. `prefix`: is the prefix path to store the volumes inside the backup tarball file
+2. `volumes`: list of volume targets inside the container, should be absolute paths to a directory or a file inside the container
```

Comment:
I feel this could be explained better. This `volumes` are the volumes from the target container but in case of bind-mount volumes what part of the volume definition should be used?

I think you can make this easier if you expand the below example:
- Add the target container definition (something along the lines of "given the following running container: <docker run -v volume_dir -v volume_file ....> the following config file <config file> would save `volume_dir` and `volume_file` in the `prefix/path` directory inside the backup tar"

Do you get my idea here? Set a full example and everything will be super easier for the user.

Reply:
I got the idea. A full example was added here https://github.com/NethermindEth/docker-volumes-snapshotter/pull/1/commits/89ee98b0715b976c409e5f52a1a8b5d9ae959511